src/plugins/intel_cpu/src/extension.cpp

```diff
@@ -2,204 +2,143 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
-#include "extension.h"
+#include <openvino/core/extension.hpp>
+#include <openvino/core/op_extension.hpp>
+#include <ov_ops/augru_cell.hpp>
+#include <ov_ops/augru_sequence.hpp>
+#include <ov_ops/multiclass_nms_ie_internal.hpp>
+#include <ov_ops/nms_ie_internal.hpp>
+#include <ov_ops/nms_static_shape_ie.hpp>
+#include <ov_ops/type_relaxed.hpp>
+
+#include "snippets/op/subgraph.hpp"
 #include "transformations/cpu_opset/common/op/fully_connected.hpp"
 #include "transformations/cpu_opset/common/op/leaky_relu.hpp"
+#include "transformations/cpu_opset/common/op/ngram.hpp"
 #include "transformations/cpu_opset/common/op/power_static.hpp"
 #include "transformations/cpu_opset/common/op/sdpa.hpp"
 #include "transformations/cpu_opset/common/op/swish_cpu.hpp"
-#include "transformations/cpu_opset/common/op/ngram.hpp"
-#include "transformations/cpu_opset/x64/op/mha.hpp"
 #include "transformations/cpu_opset/x64/op/interaction.hpp"
-#include "transformations/snippets/x64/op/load_convert.hpp"
-#include "transformations/snippets/x64/op/store_convert.hpp"
-#include "transformations/snippets/x64/op/brgemm_cpu.hpp"
+#include "transformations/cpu_opset/x64/op/mha.hpp"
 #include "transformations/snippets/x64/op/brgemm_copy_b.hpp"
+#include "transformations/snippets/x64/op/brgemm_cpu.hpp"
+#include "transformations/snippets/x64/op/load_convert.hpp"
 #include "transformations/snippets/x64/op/perf_count_rdtsc.hpp"
+#include "transformations/snippets/x64/op/store_convert.hpp"
 
-#include <ov_ops/augru_cell.hpp>
-#include <ov_ops/augru_sequence.hpp>
-#include <ov_ops/type_relaxed.hpp>
-#include <ov_ops/nms_ie_internal.hpp>
-#include <ov_ops/nms_static_shape_ie.hpp>
-#include <ov_ops/multiclass_nms_ie_internal.hpp>
-
-#include "snippets/op/subgraph.hpp"
-
-#include <mutex>
-
-namespace ov {
-namespace intel_cpu {
-
-void Extension::GetVersion(const InferenceEngine::Version*& versionInfo) const noexcept {
-    static const InferenceEngine::Version version = {
-        {1, 0},             // extension API version
-        "1.0",
-        "Extension"   // extension description message
-    };
-
-    versionInfo = &version;
-}
-
-void Extension::Unload() noexcept {}
-
-std::map<std::string, ngraph::OpSet> Extension::getOpSets() {
-    auto cpu_plugin_opset = []() {
-        ngraph::OpSet opset;
-
-#if defined(OPENVINO_ARCH_X86_64)
-#define NGRAPH_OP_X64(NAME, NAMESPACE) NGRAPH_OP(NAME, NAMESPACE)
+#define CPU_EXT                                                              \
```

Comment:
May I ask you to refactor this file a little bit. 
1. Lets have only 3 defines with operations list: 1) **CPU_EXTENSIONS** (CPU_EXT + INTERNAL_EXT), 2) **TYPE_RELAXED_EXTENSIONS**, 3) **SNIPPETS_EXTENSIONS** (SNIPPET_EXT + SNIPPETS_DEBUG_CAPS_EXT + SNIPPETS_DEBUG_CAPS_EXT_X86_64)
2. Continue using arch dependent defines to register new ops (the same as NGRAPH_OP, can use name like OP_EXTENSION, OP_EXTENSION_X64, etc) to hide `std::make_shared<ov::OpExtension` part underneath.